core/base/exTreeM/ExTreeM.h

```diff
@@ -0,0 +1,1743 @@
+          }
+        }
+        this->printMsg(
+          "Starting with findAscPaths", 0, ttk::debug::LineMode::REPLACE);
+        findAscPaths<triangulationType>(
+          saddleTriplets, maximaLocalToGlobal, saddlesLocalToGlobal, maximaIds,
+          saddle2Ids, order, descendingManifold, tempArray, saveGlobalIds,
+          triangulation, nMaxima, nSaddle2);
+        this->printMsg("Finished with findAscPaths", 1,
+                       ascTimer.getElapsedTime(), this->threadNumber_);
+
+        ttk::Timer pairTimer;
+        this->printMsg(
+          "Starting with PersistencePairs", 0, ttk::debug::LineMode::REPLACE);
+        constructPersistencePairs(persistencePairs, maximaTriplets,
+                                  saddleTriplets, maximaLocalToGlobal,
+                                  saddlesLocalToGlobal, globalMin);
+        this->printMsg("Finished with PersistencePairs", 1,
+                       pairTimer.getElapsedTime(), this->threadNumber_);
+
+        ttk::Timer mergeTreeTimer;
+        this->printMsg("Starting with mergetree computation", 0,
+                       ttk::debug::LineMode::REPLACE);
+        constructMergeTree(branches, maximaTriplets, maximaLocalToGlobal,
+                           saddlesLocalToGlobal, order);
+        this->printMsg("Finished with mergetree computation", 1,
+                       mergeTreeTimer.getElapsedTime(), this->threadNumber_);
+
+        ttk::Timer segmentationTimer;
+        this->printMsg("Starting with mergetree segmentation", 0,
+                       ttk::debug::LineMode::REPLACE);
+        constructSegmentation<triangulationType>(segmentation, isLeaf, branches,
+                                                 order, descendingManifold,
+                                                 tempArray, triangulation);
+        this->printMsg("Finished mergetree segmentation", 1,
+                       segmentationTimer.getElapsedTime(), this->threadNumber_);
+//  print the progress of the current subprocedure with elapsed time
+// transform the manifold back
+#pragma omp parallel for num_threads(this->threadNumber_)
+        for(ttk::SimplexId i = 0; i < nMaxima; i++) {
+          tempArray[maximaIds[i]] = saveGlobalIds[i];
+        }
+
+        this->printMsg("Computing extremum pairs",
+                       1, // progress
+                       localTimer.getElapsedTime(), this->threadNumber_);
+      }
+
+      // ---------------------------------------------------------------------
+      // print global performance
+      // ---------------------------------------------------------------------
+      {
+        this->printMsg(ttk::debug::Separator::L2); // horizontal '-' separator
+        this->printMsg(
+          "Complete", 1, globalTimer.getElapsedTime() // global progress, time
+        );
+        this->printMsg(ttk::debug::Separator::L1); // horizontal '=' separator
+      }
+
+      return 1; // return success
+    }
+
+  }; // ExTreeM class
+
+#else
```

Comment:
I'm unsure about this big #ifdef TTK_ENABLE_OPEN.
I'd recommend not to duplicate code, but rather, to put every #pragma omp in between these #ifdefs

Plus, I have the impression that the implementation below (without OpenMP) is not computing anything at all. Am I right?